src/main/clojure/clara/rules/durability.clj

```diff
@@ -184,12 +184,15 @@
 ;;; indexed-session-memory-state for more details.
 (defrecord MemIdx [idx])
 
+;;; Same as MemIdx but specific to internal objects, such as Token or Element.
+(defrecord InternalMemIdx [idx])
```

Comment:
+1 to the current approach, I just wanted to confirm that we were on the same page.  It seems we are.